src/agent/mod.rs

```diff
@@ -419,15 +432,18 @@ impl AgentContext {
         Ok(())
     }
 
-    #[tracing::instrument(level = "trace", skip(self))]
```

Comment:
I guess we can use `#[cfg_attr(feature="tracing", tracing::instrument)]` instead.